Allow to set hostname in server-address option

rzr opened the pull request:
IP is still supported.

This change is usefull when scripted (I use a docker container).

Style has been reformated using clang-format

Change-Id: I3cad78f34429886a2c4a7d707d2fa8a72fa9d61e
Signed-off-by: Philippe Coval <p.coval@samsung.com>

suyambulingamrm commented on mojo/edk/embedder/tcp_platform_handle_utils_posix.cc:
```diff
@@ -44,8 +44,22 @@ ScopedPlatformHandle CreateTCPSocket(bool needs_connection, int protocol) {
+      status = getnameinfo(result->ai_addr, result->ai_addrlen, hostname,
+                           NI_MAXHOST, NULL, 0, NI_NUMERICHOST);
+      if (status == 0 && *hostname) {
+        server_address = std::string(hostname);
```
My question may be naive. We are updating server_address to host name. But later we use inet_addr(server_address).
As per https://linux.die.net/man/3/inet_addr, it states inet_addr accepts numbers-and-dots notation as input.
Should that be ok ?

rzr commented:
Ok hostname variable name was missleading I renamed it to host
and updated commit log:

eck if server-address is a valid Internet address using getaddrinfo,
then try to resolv IP from given address in IP or hostname form.

suyambulingamrm commented:
Thanks for explaining.
Looks good to me.

venumusham commented:
lgtm

rzr commented:
This can also land in castanets branch:
https://github.com/Samsung/Castanets/pull/41

More changes to come there and doc will be updated in master branch too
